app/src/pages/inside/dashboardItemPage/modals/common/widgetControls/controls/dashboardControl/dashboardControl.jsx

```diff
@@ -12,44 +13,57 @@ const messages = defineMessages({
   },
 });
 
-export const DashboardControl = injectIntl(({ dashboards, onChange, value, intl }) => {
-  const newDashboards = [...dashboards];
-  if (newDashboards.length === 0) {
-    newDashboards.push(value);
-  }
-  return (
-    <div className={cx('dashboardControl')}>
-      <h5 className={cx('dashboardControl-title')}>
-        {intl.formatMessage(messages.dashboardControlTitle)}
-      </h5>
-      <div className={cx('dashboardControl-list')}>
-        {newDashboards.map((dashboard, index) => {
-          const active = value.id ? value.id === dashboard.id : index === 0;
-          const onClick = () => {
-            onChange(dashboard);
-          };
-          return (
-            <DashboardItem
-              key={dashboard.name}
-              active={active}
-              dashboard={dashboard}
-              onClick={onClick}
-            />
-          );
-        })}
-      </div>
-    </div>
-  );
-});
+@injectIntl
+export class DashboardControl extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    dashboards: PropTypes.arrayOf(PropTypes.object),
+    onChange: PropTypes.func,
+    value: PropTypes.object,
+  };
+
+  static defaultProps = {
+    onChange: () => {},
+    dashboards: [],
+    value: {},
+  };
+
+  render() {
+    const {
+      intl: { formatMessage },
+      dashboards,
+      onChange,
+      value,
+    } = this.props;
 
-DashboardControl.propTypes = {
-  dashboards: PropTypes.arrayOf(PropTypes.object),
-  onChange: PropTypes.func,
-  value: PropTypes.object,
-  intl: intlShape.isRequired,
-};
-DashboardControl.defaultProps = {
-  onChange: () => {},
-  dashboards: [],
-  value: {},
-};
+    const newDashboards = [...dashboards];
+
+    if (newDashboards.length === 0) {
+      newDashboards.push(value);
+    }
+
+    return (
+      <div className={cx('dashboardControl')}>
+        <h5 className={cx('dashboardControl-title')}>
+          {formatMessage(messages.dashboardControlTitle)}
+        </h5>
+        <div className={cx('dashboardControl-list')}>
+          {newDashboards.map((dashboard, index) => {
+            const active = value.id ? value.id === dashboard.id : index === 0;
+            const onClick = () => {
```

Comment:
This handler pretty simple for create a variable for it - just put it into `onClick` prop: `onClick={() => onChange(dashboard)}`.

Reply:
Fixed